pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kubewharf/katalyst-api/pkg/plugins/registration"
+	"github.com/kubewharf/katalyst-api/pkg/plugins/skeleton"
+
+	pluginapi "github.com/kubewharf/katalyst-api/pkg/protocol/evictionplugin/v1alpha1"
+
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/poweraware/evictor"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+)
+
+const (
+	EvictionPluginNameNodePowerPressure = "node_power_pressure"
+	evictReason                         = "host under power pressure"
+)
+
+type powerPressureEvictServer struct {
+	mutex   sync.RWMutex
+	evicts  map[types.UID]*v1.Pod
+	service *skeleton.PluginRegistrationWrapper
+}
+
+func (p *powerPressureEvictServer) Init() error {
+	return nil
+}
+
+// reset method clears all pending eviction requests not fetched by remote client
+func (p *powerPressureEvictServer) reset(ctx context.Context) {
+	p.evicts = make(map[types.UID]*v1.Pod)
+}
+
+// Evict method puts request to evict pods in the pool; it will be sent out to plugin client via the eviction protocol
+// the real eviction will be done by the (remote) eviction manager where the plugin client is registered with
+func (p *powerPressureEvictServer) Evict(ctx context.Context, pods []*v1.Pod) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	// discard pending requests not handled yet; we will provide a new sleet of evict requests anyway
+	p.reset(ctx)
+
+	for _, pod := range pods {
+		if err := p.evictPod(ctx, pod); err != nil {
+			return errors.Wrap(err, "failed to put evict pods to the service pool")
+		}
+	}
+
+	return nil
+}
+
+func (p *powerPressureEvictServer) evictPod(ctx context.Context, pod *v1.Pod) error {
+	if pod == nil {
+		return errors.New("unexpected nil pod")
+	}
+
+	p.evicts[pod.GetUID()] = pod
+	return nil
+}
+
+func (p *powerPressureEvictServer) Name() string {
+	return EvictionPluginNameNodePowerPressure
+}
+
+func (p *powerPressureEvictServer) Start() error {
+	if err := p.service.Start(); err != nil {
+		return errors.Wrap(err, "failed to start power pressure eviction plugin server")
+	}
+	return nil
+}
+
+func (p *powerPressureEvictServer) Stop() error {
+	return nil
```

Comment:
Do we need to stop `p.service`?

Reply:
yes having a Stop method sounds good; it helps to provide meaningful error message for Evict calls